src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -746,14 +883,90 @@ struct MHASingleToken {
                     const PlainTensor& attention_mask,
                     PlainTensor& output_emb,
                     const PlainTensor& beams,
+                    size_t max_context_len,
                     const PlainTensor& context_lens,
                     bool has_out_transpose,
                     bool auto_causal,
                     float d_scale,
                     const PlainTensor& k_scale_zp,
                     const PlainTensor& v_scale_zp) {
-        mha_single_token(query, present_key, present_value, alibi_mask, attention_mask, beams, context_lens, output_emb,
-            m_attn_w, m_temp, has_out_transpose, auto_causal, d_scale, k_scale_zp, v_scale_zp, m_head_sum);
+        auto B = query.size(0);
+        auto H = query.size(1);
+        auto q_len = query.size(2);
+        bool is_pagedattn = context_lens;
+        size_t kv_len;
+        if (is_pagedattn) {
+            kv_len = max_context_len;
+        } else {
+            kv_len = present_key.size(2);
+        }
+
+        bool fastpath_valid = false;
+#ifdef OPENVINO_ARCH_X86_64
+        if (is_pagedattn) {
+            auto S = query.size(3);
+            size_t block_size = present_value.size(2);
+            fastpath_valid = mayiuse(amx_bf16) && (S % 32 == 0) && (block_size % 16 == 0) && (S <= 32 * 6) && present_key.get_precision() == ov::element::bf16;
+            if (fastpath_valid) {
+                m_attn_w.resize<float>({B, H, q_len, (kv_len + block_size - 1) / block_size * block_size});
+                if (!m_gemv)
+                    m_gemv = std::make_shared<JitMatMulVecAMX>(static_cast<int>(S), static_cast<int>(block_size));
```

Comment:
Is `block_size` a constant value during inference?

Reply:
Yes, when the compiled model is created, it's a constant.